### PR TITLE
Issue #SC-1924 feat: api on-boarding script

### PR DESCRIPTION
**Pull request by krgauraw**

### CODE REVIEW

File: ansible/roles/kong-api/defaults/main.yml [hunk 1]
```diff
@@ -5164,3 +5167,110 @@ kong_apis:
...
+    uris: "{{ object_category_prefix }}/v1/read"
+    upstream_url: "{{ taxonomy_service_url }}/object/category/v4/read"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'frameworkCreate'
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5198]: change the ACL to read group and not create @krgauraw .

File: ansible/roles/kong-api/defaults/main.yml [hunk 2]
```diff
@@ -5164,3 +5167,110 @@ kong_apis:
...
+    uris: "{{ object_category_prefix }}/v1/update"
+    upstream_url: "{{ taxonomy_service_url }}/object/category/v4/update"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'frameworkCreate'
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5216]: change the ACL to update group and not create @krgauraw .

File: ansible/roles/kong-api/defaults/main.yml [hunk 3]
```diff
@@ -5164,3 +5167,110 @@ kong_apis:
...
+    uris: "{{ object_category_definition_prefix }}/v1/read"
+    upstream_url: "{{ taxonomy_service_url }}/object/category/definition/v4/read"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'frameworkCreate'
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5252]: change the ACL to read group and not create @krgauraw .

File: ansible/roles/kong-api/defaults/main.yml [hunk 4]
```diff
@@ -5164,3 +5167,110 @@ kong_apis:
...
+    uris: "{{ object_category_definition_prefix }}/v1/update"
+    upstream_url: "{{ taxonomy_service_url }}/object/category/definition/v4/update"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'frameworkCreate'
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5270]: change the ACL to update group and not create @krgauraw .

File: ansible/roles/kong-api/defaults/main.yml [hunk 5]
```diff
@@ -5164,3 +5167,110 @@ kong_apis:
...
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ x_medium_rate_limit_per_hour }}"
+      config.limit_by: credential
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5184]: will these APIs be used by mobile devices as well?

File: ansible/roles/kong-api/defaults/main.yml [hunk 6]
```diff
@@ -5164,3 +5167,110 @@ kong_apis:
...
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'frameworkCreate'
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ x_medium_rate_limit_per_hour }}"
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5183]: provide small rate limit per hour @krgauraw

File: ansible/roles/kong-api/defaults/main.yml [hunk 7]
```diff
@@ -5164,3 +5167,110 @@ kong_apis:
...
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5219]: provide small rate limit per hour for this api as well @krgauraw

File: ansible/roles/kong-api/defaults/main.yml [hunk 8]
```diff
@@ -5164,3 +5167,110 @@ kong_apis:
...
+      config.limit_by: credential
+    - name: request-size-limiting
+      config.allowed_payload_size: "{{ small_request_size_limit }}"
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5240]: is the small payload size sufficient for object category definition @krgauraw ?

File: ansible/roles/kong-api/defaults/main.yml (hunk 5 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml:5184]: no

File: ansible/roles/kong-api/defaults/main.yml (hunk 7 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml:5219]: what is the number assigned with small rate ? @SMY-ALTAMASH

File: ansible/roles/kong-api/defaults/main.yml (hunk 8 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml:5240]: yes. thats sufficient

File: ansible/roles/kong-api/defaults/main.yml (hunk 1 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml]: done

File: ansible/roles/kong-api/defaults/main.yml (hunk 2 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml]: done

File: ansible/roles/kong-api/defaults/main.yml (hunk 3 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml]: done

File: ansible/roles/kong-api/defaults/main.yml (hunk 6 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml]: done

File: ansible/roles/kong-api/defaults/main.yml (hunk 7 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml]: done.

File: ansible/roles/kong-api/defaults/main.yml (hunk 4 above)
Comment (krgauraw) [reply ansible/roles/kong-api/defaults/main.yml]: done

File: ansible/roles/kong-api/defaults/main.yml (hunk 5 above)
Comment (SMYALTAMASH) [reply ansible/roles/kong-api/defaults/main.yml:5184]: then change the value to consumer for limit_by

**Review by SMYALTAMASH — APPROVED** [review]

**Pull request merged**